src/DurableTask.SqlServer/Scripts/logic.sql

```diff
@@ -605,13 +612,38 @@ BEGIN
         SUBSTRING(E.[InstanceID], 2, CHARINDEX('@', E.[InstanceID], 2) - 2),
         'Pending'
     FROM @NewOrchestrationEvents E
-    WHERE NOT EXISTS (
-        SELECT 1
-        FROM dt.Instances I
-        WHERE [TaskHub] = @TaskHub AND I.[InstanceID] = E.[InstanceID])
+    WHERE LEFT(E.[InstanceID], 1) = '@'
+        AND CHARINDEX('@', E.[InstanceID], 2) > 0
```

Comment:
I had to add this limiting to avoid errors when invoking SUBSTRING function on instance ids for sub-orchestrations, since they do not begin with or contain '@' symbol.